apps/opik-frontend/src/components/pages/PromptPage/ExperimentsTab/ExperimentsTab.tsx

```diff
@@ -103,12 +132,69 @@ const ExperimentsTab: React.FC<ExperimentsTabProps> = ({ promptId }) => {
     ? "There are no experiments used this prompt"
     : "No search results";
 
+  const [selectedColumns, setSelectedColumns] = useLocalStorageState<string[]>(
+    SELECTED_COLUMNS_KEY,
+    {
+      defaultValue: DEFAULT_SELECTED_COLUMNS,
+    },
+  );
+
+  const [, setPresentedDynamicColumns] = useLocalStorageState<string[]>(
+    DYNAMIC_COLUMNS_KEY,
+    {
+      defaultValue: [],
+    },
+  );
+
+  const [columnsOrder, setColumnsOrder] = useLocalStorageState<string[]>(
+    COLUMNS_ORDER_KEY,
+    {
+      defaultValue: [],
+    },
+  );
+
+  const [scoresColumnsOrder, setScoresColumnsOrder] = useLocalStorageState<
+    string[]
+  >(COLUMNS_SCORES_ORDER_KEY, {
+    defaultValue: [],
+  });
+
   const [columnsWidth, setColumnsWidth] = useLocalStorageState<
     Record<string, number>
   >(COLUMNS_WIDTH_KEY, {
     defaultValue: {},
   });
 
+  useEffect(() => {
+    setPresentedDynamicColumns((cols) => {
+      const dynamicColumnsIds = dynamicColumns.map((col) => col.id);
```

Comment:
maybe, you could put this logic into some function / hook. the logic here is getting harder to follow and it starts being repeated in a few files

Reply:
Good cache. I also noticed this. I will move this logic to a generic location when adding more dynamic columns across the application.